cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -559,6 +564,24 @@ var (
 			rawUnsetPaths, _ := cmd.Flags().GetStringSlice("unset")
 			unsetPaths := util.NormalizePaths(rawUnsetPaths)
 
+			var device ttnpb.EndDevice
+			if hasUpdateDeviceLocationFlags(cmd.Flags()) {
+				is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
```

Comment:
Do we really need to dial and fetch the existing device twice? We already do this on L609 (L631 in your PR).